nav2_costmap_2d/src/costmap_2d_publisher.cpp

```diff
@@ -115,13 +117,20 @@ void Costmap2DPublisher::onNewSubscription(const ros::SingleSubscriberPublisher&
   pub.publish(grid_);
 } */
 
-// prepare grid_ message for publication.
-void Costmap2DPublisher::prepareGrid()
+
+void Costmap2DPublisher::updateGridParams()
 {
-  std::unique_lock<Costmap2D::mutex_t> lock(*(costmap_->getMutex()));
+  saved_origin_x_ = costmap_->getOriginX();
+  saved_origin_y_ = costmap_->getOriginY();
   grid_resolution = costmap_->getResolution();
```

Comment:
Grid resolution/width/height should have trailing underscores as class member variables

Reply:
resolved with 9b92028155ceefbfa42881738e212f44109a26fe